cmake/vif.cmake

```diff
@@ -5,13 +5,17 @@
 set(gen_vif_script ${ZEPHYR_BASE}/scripts/generate_usb_vif/generate_vif.py)
 set(dts_compatible usb-c-connector)
 set(vif_xml ${PROJECT_BINARY_DIR}/vif.xml)
+
 set(cmd_gen_vif ${PYTHON_EXECUTABLE} ${gen_vif_script}
         --edt-pickle ${EDT_PICKLE}
         --compatible ${dts_compatible}
         --vif-out ${vif_xml}
-        --board ${BOARD}
         )
 
+if(EXISTS ${APPLICATION_SOURCE_DIR}/vif_${BOARD}.xml)
+    list(APPEND cmd_gen_vif --vif-source-xml ${APPLICATION_SOURCE_DIR}/vif_${BOARD}.xml)
```

Comment:
This looks like it would get messy with files in the root directory of e.g. a sample - there already is a place for board overlays to go in the `boards` folder. I don't know about vif but is this something you would only ever define at the application level i.e. can this also be defined at a board level just once as opposed to once per board per application, which quickly grows out of control with multiple samples?

Reply:
I got confused/missed thinking about scope of VIF, thank you for pointing out. In the Zephyr sample app "sink", we need different source VIF per board, but there is use-case in ChromeOS where there could be multiple apps using a board(i.e, chip) requires different VIF at application level but not at board. I am evaluating different approaches to be able to have choice to have the VIF in either at application level or board level, will discuss with you offline regarding them. Thank you

Reply:
Hi, I updated the script to look in both application and board directories. Apart from that a user could configure the file location through Kconfig options if they want to store in different folder with different name in application or board source directory. Thank you